core/src/main/resources/META-INF/native-image/com.datastax.oss/java-driver-core/reflection.json

```diff
@@ -0,0 +1,175 @@
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.specex.NoSpeculativeExecutionPolicy",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.metadata.NoopNodeStateListener",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.tracker.NoopRequestTracker",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.session.throttling.PassThroughRequestThrottler",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.time.AtomicTimestampGenerator",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.channel.InFlightHandler",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.channel.ProtocolInitHandler",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.protocol.FrameDecoder",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.protocol.FrameEncoder",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.channel.ChannelFactory$ChannelFactoryInitializer",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.channel.HeartbeatHandler",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "com.datastax.oss.driver.internal.core.metrics.DefaultMetricsFactory",
+    "allDeclaredConstructors": true,
+    "allPublicConstructors": true,
+    "allDeclaredMethods": true,
+    "allPublicMethods": true
+  },
+  {
+    "name": "io.netty.channel.socket.nio.NioSocketChannel",
+    "methods": [
+      { "name": "<init>", "parameterTypes": [] }
+    ]
+  },
```

Comment:
Without this exclusion:

```
Exception in thread "main" java.lang.IllegalArgumentException: Class NioSocketChannel does not have a public non-arg constructor                                                  
        at io.netty.channel.ReflectiveChannelFactory.<init>(ReflectiveChannelFactory.java:36)                                                                                     
        at io.netty.bootstrap.AbstractBootstrap.channel(AbstractBootstrap.java:110)                                                                                               
        at com.datastax.oss.driver.internal.core.channel.ChannelFactory.connect(ChannelFactory.java:192)                                                                          
        at com.datastax.oss.driver.internal.core.channel.ChannelFactory.connect(ChannelFactory.java:167)                                                                          
        at com.datastax.oss.driver.internal.core.channel.ChannelFactory.connect(ChannelFactory.java:148)                                                                          
        at com.datastax.oss.driver.internal.core.control.ControlConnection$SingleThreaded.connect(ControlConnection.java:363)                                                     
        at com.datastax.oss.driver.internal.core.control.ControlConnection$SingleThreaded.init(ControlConnection.java:303)                                                        
        at com.datastax.oss.driver.internal.core.control.ControlConnection$SingleThreaded.access$1100(ControlConnection.java:243)                                                 
        at com.datastax.oss.driver.internal.core.control.ControlConnection.lambda$init$0(ControlConnection.java:122)                                                              
        at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)                                                                                                      
        at io.netty.util.concurrent.PromiseTask.run(PromiseTask.java:106)                                                                                                         
        at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54)                                                                                                        
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)                                                                           
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)                                                                                              
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)                                                                                  
        at java.lang.Thread.run(Thread.java:748)                                                                                                                                  
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:553)                                                                                        
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)                                                                       
Caused by: java.lang.NoSuchMethodException: io.netty.channel.socket.nio.NioSocketChannel.<init>()                                                                                 
        at java.lang.Class.getConstructor0(DynamicHub.java:3082)                                                                                                                  
        at java.lang.Class.getConstructor(DynamicHub.java:1825)                                                                                                                   
        at io.netty.channel.ReflectiveChannelFactory.<init>(ReflectiveChannelFactory.java:34)
        ... 17 more
```

Not abundantly clear why this would be; NioSocketChannel very definitely does have a no-arg constructor.  It's possible there's another Graal substitution at play here... I just don't know.

Note that Helidon had to address something similar: see https://github.com/oracle/helidon/pull/1722